internal/semerrgroup/sem.go

```diff
@@ -30,6 +44,29 @@ func (s *Group) Go(fn func() error) {
 }
 
 // Wait waits for the group to complete and return an error if any.
-func (s *Group) Wait() error {
+func (s *parallelGroup) Wait() error {
 	return s.g.Wait()
 }
+
+var _ Group = &serialGroup{}
+
+type serialGroup struct {
+	lock sync.Mutex
+	err  error
+}
+
+// Go execs one function at a time.
+func (s *serialGroup) Go(fn func() error) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
```

Comment:
do we need to lock here?

since we don't have any errgroup or whatever, it seems like it won't really do anything...

Reply:
it serializes access to err in case this is called in parallel. if it's not called in parallel anywhere, I can remove this. What do you think?

Reply:
you can remove it yes... :)